test/integration/cli/generate_test.rb

```diff
@@ -81,12 +81,24 @@ def after
       before do
         generate_model
       end
+      describe 'when model names are Underscored names.' do
+        it 'generates model' do
+          @root.join('lib/delivery/entities/test.rb').must_be                      :exist?
+          @root.join('lib/delivery/repositories/test_repository.rb').must_be       :exist?
+          @root.join('spec/delivery/entities/test_spec.rb').must_be                :exist?
+          @root.join('spec/delivery/repositories/test_repository_spec.rb').must_be :exist?
+        end
+      end
 
-      it 'generates model' do
-        @root.join('lib/delivery/entities/test.rb').must_be                      :exist?
-        @root.join('lib/delivery/repositories/test_repository.rb').must_be       :exist?
-        @root.join('spec/delivery/entities/test_spec.rb').must_be                :exist?
-        @root.join('spec/delivery/repositories/test_repository_spec.rb').must_be :exist?
+      describe 'when model names are CamelCase names.' do
+        let(:klass) { 'Test' }
```

Comment:
Instead `Test` can be `TestCase` because with this test you assert correct camelcase an underscore